toolkit-test-app/src/main/java/com/esri/arcgisruntime/toolkit/test/ar/ArcGISArViewActivity.kt

```diff
@@ -303,11 +312,28 @@ class ArcGISArViewActivity : AppCompatActivity() {
             override fun onSingleTapConfirmed(motionEvent: MotionEvent?): Boolean {
                 motionEvent?.let {
                     with(Point(motionEvent.x.toInt(), motionEvent.y.toInt())) {
-                        if (arcGisArView.locationDataSource != null) {
-                            // TODO implement creation of point when implementing screenToLocation
-                        } else {
+                        if (currentScene?.isTabletop == true) {
                             arcGisArView.setInitialTransformationMatrix(this)
+                        } else {
+                            var sphere = SimpleMarkerSceneSymbol.createSphere(
+                                Color.CYAN,
+                                0.25,
+                                SceneSymbol.AnchorPosition.BOTTOM
+                            )
+                            arcGisArView.arScreenToLocation(this)?.let {
+                                if (sphereOverlay == null) {
```

Comment:
Could you make `sphereOverlay` a lazy property and have all of this logic of creating/adding the GraphicsOverlay in the property implementation?

Reply:
🤦‍♂ yeah. I don't know why I did it this way.